src/main/java/seedu/address/model/interview/UniqueInterviewList.java

```diff
@@ -50,20 +52,25 @@ public void add(Interview toAdd) {
         internalList.add(toAdd);
     }
     /**
-     * Replaces the candidate {@code target} in the list with {@code editedCandidate}.
+     * Replaces the interview {@code target} in the list with {@code editedInterview}.
      * {@code target} must exist in the list.
-     * The candidate identity of {@code editedCandidate} must not be the same as another existing candidate in the list.
+     * The interview {@code editedInterview} must not be in conflict with another interview for a different candidate.
      */
-    public void setInterview(Interview target, Interview editedInterview) {
+    public void setInterview(Interview target, Interview editedInterview) throws CommandException {
         requireAllNonNull(target, editedInterview);
 
         int index = internalList.indexOf(target);
         if (index == -1) {
             throw new InterviewNotFoundException();
         }
 
-        if (containsConflictingInterview(editedInterview)) {
-            throw new ConflictingInterviewException();
+        ObservableList<Interview> internalListCopy = FXCollections.observableArrayList(internalList);
+        // Credits to teammate @tiewweijian for initial suggestion of deletion/removal of interviews
+        // to solve the issue of rescheduling the same candidate
+        internalListCopy.set(index, editedInterview);
+
+        if (!interviewsDateTimeAreNonConflicting(internalListCopy)) {
+            throw new CommandException(MESSAGE_CONFLICTING_INTERVIEW);
```

Comment:
This looks like a cleaner and probably more defensive way of implementing it ! Great job @lzan98 !!